src/webui/www/private/newtag.html

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
+<html lang="${LANG}">
+
+<head>
+    <meta charset="UTF-8" />
+    <title>QBT_TR(Add Tags)QBT_TR[CONTEXT=TransferListWidget]</title>
+    <link rel="stylesheet" href="css/style.css?v=${CACHEID}" type="text/css" />
+    <script src="scripts/lib/mootools-1.2-core-yc.js"></script>
+    <script src="scripts/lib/mootools-1.2-more.js"></script>
+    <script src="scripts/misc.js?locale=${LANG}&v=${CACHEID}"></script>
+    <script>
+        'use strict';
+
+        new Keyboard({
+            defaultEventType: 'keydown',
+            events: {
+                'Enter': function(event) {
+                    $('tagNameButton').click();
+                    event.preventDefault();
+                },
+                'Escape': function(event) {
+                    window.parent.closeWindows();
+                    event.preventDefault();
+                },
+                'Esc': function(event) {
+                    window.parent.closeWindows();
+                    event.preventDefault();
+                }
+            }
+        }).activate();
+
+        window.addEvent('domready', function() {
+            const uriAction = safeTrim(new URI().getData('action'));
+            const uriHashes = safeTrim(new URI().getData('hashes'));
+
+            if (uriAction === 'create')
+                $('legendText').innerText = 'QBT_TR(Tag:)QBT_TR[CONTEXT=TagFilterWidget]';
+
+            $('tagName').focus();
+
+            $('tagNameButton').addEvent('click', function(e) {
+                new Event(e).stop();
+
+                const tagName = $('tagName').value.trim();
+
+                const verifyTagName = function(name) {
+                    if ((name === null) || (name === ""))
+                        return false;
+                    if (name.indexOf(",") >= 0) {
+                        alert("QBT_TR(Invalid tag name)QBT_TR[CONTEXT=TagFilterWidget]");
+                        return false;
+                    }
+                    return true;
+                };
+
+                switch (uriAction) {
+                    case "set":
```

Comment:
How do I evoke this case from the WebUI? It seems set just calls `api/v2/torrents/addTags` or `api/v2/torrents/removeTags` directly from `mocha-init.js`, rather than opening this dialog.

Reply:
You can do this via torrent context menu -> Tags -> Add...

Reply:
That did it 👍  thanks. I wrote the original code for this with add categories and it *still* confuses me. Probably not a sign of good code (on my part).